lib/modules/platform/github/massage-markdown-links.ts

```diff
@@ -11,12 +11,19 @@ interface UrlMatch {
 }
 
 const urlRegex =
-  /(?:https?:)?(?:\/\/)?(?:www\.)?(?<!api\.)(?:to)?github\.com\/[-_a-z0-9]+\/[-_a-z0-9]+\/(?:discussions|issues|pull)\/[0-9]+(?:#[-_a-z0-9]+)?/i; // TODO #12872 (?<!re) after text not matching
+  /(?:https?:)?(?:\/\/)?(?:www\.)?(?<!api\.)(?:to)?github\.com\/[-_a-z0-9]+\/[a-zA-Z1-9\-_.]+\/(?:discussions|issues|pull)\/[0-9]+(?:#[-_a-z0-9]+)?/i; // TODO #12872 (?<!re) after text not matching
```

Comment:
@viceice if we look at the changes here, this is the only change in the middle:
\/[a-zA-Z1-9\-_.]+\/
\/[-_a-z0-9]+\/ 
you approved this so hopefully you can explain to me
is it correct to remove capital letters backward slashes and dots(which not skipped for some reason)  here? is the URL always with small letters? or is it just so we won't catch any user names that have capital letters? 
the intention isn't clear to me here.

Reply:
nevermind, there's insensitive flag at the end it means no need for big letters...

Reply:
😉  allowed usernames[^1]

[^1]: https://github.com/shinnn/github-username-regex